packages/compass-indexes/src/components/search-indexes-modals/base-search-index-modal.tsx

```diff
@@ -0,0 +1,191 @@
+import { useTrackOnChange } from '@mongodb-js/compass-logging';
+import {
+  Modal,
+  ModalFooter,
+  ModalHeader,
+  ModalBody,
+  TextInput,
+  Label,
+  spacing,
+  css,
+  HorizontalRule,
+  Subtitle,
+  Button,
+  Link,
+  Icon,
+  WarningSummary,
+  ErrorSummary,
+} from '@mongodb-js/compass-components';
+import { CodemirrorMultilineEditor } from '@mongodb-js/compass-editor';
+import _parseShellBSON, { ParseMode } from 'ejson-shell-parser';
+import type { Document } from 'mongodb';
+
+// Copied from packages/compass-aggregations/src/modules/pipeline-builder/pipeline-parser/utils.ts
+function parseShellBSON(source: string): Document[] {
+  const parsed = _parseShellBSON(source, { mode: ParseMode.Loose });
+  if (!parsed || typeof parsed !== 'object') {
+    // XXX(COMPASS-5689): We've hit the condition in
+    // https://github.com/mongodb-js/ejson-shell-parser/blob/c9c0145ababae52536ccd2244ac2ad01a4bbdef3/src/index.ts#L36
+    throw new Error('The provided index definition is invalid.');
+  }
+  return parsed;
+}
+
+const flexWithGapStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: spacing[3],
+});
+
+const bodyGapStyles = css({
+  marginTop: spacing[3],
+});
+
+const toolbarStyles = css({
+  display: 'flex',
+  gap: spacing[2],
+});
+
+type BaseSearchIndexModalProps = {
+  title: string;
+  submitActionName: string;
+  initialIndexName: string;
+  initialIndexDefinition: string;
+  isIndexNameReadonly: boolean;
+  isModalOpen: boolean;
+  isBusy: boolean;
+  error: string | undefined;
+  submitIndex: (indexName: string, indexDefinition: Document) => void;
+  closeModal: () => void;
+};
+
+export const BaseSearchIndexModal: React.FunctionComponent<
+  BaseSearchIndexModalProps
+> = ({
+  title,
+  submitActionName,
+  initialIndexName,
+  isIndexNameReadonly,
+  initialIndexDefinition,
+  isModalOpen,
+  isBusy,
+  error,
+  submitIndex,
+  closeModal,
+}) => {
+  const [indexName, setIndexName] = useState<string>(initialIndexName);
+  const [indexDefinition, setIndexDefinition] = useState<string>(
+    () => initialIndexDefinition
+  );
+  const [parsingError, setParsingError] = useState<string | undefined>(
+    undefined
+  );
+
+  const onSearchIndexDefinitionChanged = useCallback(
+    (newDefinition: string) => {
+      setParsingError(undefined);
+
+      try {
+        parseShellBSON(newDefinition);
+        setIndexDefinition(newDefinition);
+      } catch (ex) {
+        setParsingError((ex as Error).message);
+      }
+    },
+    [setIndexDefinition, setParsingError]
+  );
+
+  const onSubmitIndex = useCallback(() => {
+    if (parsingError) {
+      return;
+    }
+
+    const indexDefinitionDoc = parseShellBSON(indexDefinition);
+    submitIndex(indexName, indexDefinitionDoc);
+  }, [submitIndex, parsingError, indexName, indexDefinition]);
+
+  useTrackOnChange(
+    'COMPASS-SEARCH-INDEXES-UI',
+    (track) => {
+      if (isModalOpen) {
+        track('Screen', { name: 'create_search_index_modal' });
```

Comment:
This will always track as `create` and not update. I don't think we have existing event for update (regular) index, but probably need to introduce one. 

(Or you can completely remove it in this PR and then we can handle this in [COMPASS-7170](https://jira.mongodb.org/browse/COMPASS-7170))

Reply:
I'll remove it entirely then, so we can implement it properly in COMPASS-7170.